inc/saitam.h

```diff
@@ -0,0 +1,883 @@
+        _In_ uint32_t count,
+        _In_ sai_tam_threshold_breach_event_t *data);
+
+/**
+ * @brief TAM Threshold Attributes.
+ */
+typedef enum _sai_tam_threshold_attr_t
+{
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_THRESHOLD_ATTR_START,
+
+    /**
+     * @brief TAM Object
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE
+     */
+    SAI_TAM_THRESHOLD_ATTR_TAM_ID = SAI_TAM_THRESHOLD_ATTR_START,
+
+    /**
+     * @brief Statistic for this threshold
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE
+     * @objects SAI_OBJECT_TYPE_TAM_STAT
+     */
+    SAI_TAM_THRESHOLD_ATTR_STATISTIC,
+
+    /**
+     * @brief Threshold Level
+     * Breach level for this threshold in number of bytes
+     *
+     * If specified, a threshold breach event will be recorded when the buffer
+     * usage goes beyond the level.
+     *
+     * If not specified, then by default the threshold is created without
+     * any level, which is effectively disabling the threshold monitoring
+     * for the statistic
+     *
+     * @type sai_uint64_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_TAM_THRESHOLD_ATTR_LEVEL,
+
+    /**
+     * @brief Transporter Object
+     *
+     * Provides the snapshot transporter object for this threshold.
+     * When the threshold is breached, this transporter will be used
+     * to 'copy' the data to the 'transporter-desired' location.
+     *
+     * In the absence of a transporter, the tracker's default transporter
+     * will be used.
+     *
+     * It may be noted that, Upon a breach, the 'snapshot' object is
+     * automatically created (see below attribute), and it will not have a
+     * separate transporter object. Instead this transporter (or the tracker's
+     * default transporter) is used.
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_TRANSPORTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_TAM_THRESHOLD_ATTR_TRANSPORTER,
+
+    /**
+     * @brief Snapshot on breach
+     *
+     * Take a snapshot upon a threshold breach.
+     * When this attribute is specified and set to true, Snapshots are
+     * automatically created upon the threshold breach.
+     * Otherwise a snapshot is not created.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_TAM_THRESHOLD_ATTR_SNAPSHOT_ON_BREACH,
+
+    /**
+     * @brief Buffers/Statistics for inclusion in the snapshot
+     * Specifies the Statistics/Types for the snapshot.
+     * If not specified, all buffers tracked by
+     * the associated TAM object are included in the snapshot.
+     * When specified, the buffers requested for snapshot must be within the set
+     * tracked by the associated TAM object.
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_TAM_STAT
```

Comment:
@tushar-ty , since this is CREATE_AND_SET, this attribute is missing default value, can you add.